### PR TITLE
Run all instrumentation tests on CI and fix media player illegal state

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

The PR:
- Refactors the CI pipeline to run more jobs in parallel, cutting down on the execution time.
- Enables `Timber` logging in the instrumentation tests and refactors the banner instructions test.
- Enables the Test Orchestrator to ensure that each test runs in its own runner instance, ruling out most of the async operations leaks from previous test cases.

The test suite automation introduced in this PR was able to consistently reproduce https://github.com/mapbox/mapbox-navigation-android/issues/3750, so this PR is basically dependent on https://github.com/mapbox/mapbox-navigation-android/pull/3891, otherwise, the tests are really flaky.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -242,20 +255,20 @@ commands:
...
-        description: module target
+        description: SDK module target
         type: string
       variant:
         description: debug or release
         type: string
     steps:
       - run:
-          name: Run instrumentation tests on Firebase
+          name: Run instrumentation tests on intenral SDK classes on Firebase
```
Comment (Guardiola31337) [inline circle.yml:271]:
```suggestion
          name: Run instrumentation tests on internal SDK classes on Firebase
```

**Review by Guardiola31337 — COMMENTED** [review]
> Refactors the CI pipeline to run more jobs in parallel, cutting down on the execution time.

Out of curiosity, do we have a quick benchmark comparison of the before and after?

> Enables the Test Orchestrator to ensure that each test runs in its own runner instance, ruling out most of the async operations leaks from previous test cases.

Did we have these async operations leaks before? Not sure if this is ultimately going to rule out all of them as this was enabled before https://github.com/mapbox/mapbox-navigation-android/blob/dcad00fe7ad0ba9aa50fcf3d3aac6accf40e191d/examples/build.gradle#L42-L44 🤔 

We need to tweak Github `Required` checks? BTW, what are those now?

File: circle.yml (hunk 1 above)
Comment (LukasPaczos) [reply circle.yml:271]: Adjust next time? :)

Don't want to rerun the whole suite for this typo.

**Comment by LukasPaczos** [comment]
> Out of curiosity, do we have a quick benchmark comparison of the before and after?

Before (~39 mins):
https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/9957/workflows/d1f43e43-356b-4550-aadd-f49a5a43432c
After (~24 mins):
https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-android/9958/workflows/c5e57c22-7eb4-4f4c-99ed-422cb799776f

> Not sure if this is ultimately going to rule out all of them as this was enabled before

It was only enabled for the `examples` module which was not actually running any instrumentation tests. There is no legacy suite to compare to.

> We need to tweak Github Required checks? BTW, what are those now?

I'll do that right before merging the PR, after it's accepted.

**Comment by Guardiola31337** [comment]
> It was only enabled for the `examples` module which was not actually running any instrumentation tests. There is no legacy suite to compare to.

We had some instrumentation tests in the past that run through the `examples` module. Anyways 🚀

**Review by Guardiola31337 — APPROVED** [review]

File: circle.yml (hunk 1 above)
Comment (Guardiola31337) [reply circle.yml:271]: https://github.com/mapbox/mapbox-navigation-android/pull/3895/commits/f036f1115fd33a48d7c22a4e8693b2a96251ba73

**Pull request merged**